app/routes/orders/contact_summary.js

```diff
@@ -1,10 +1,26 @@
-import detail from './detail';
+import detail from "./detail";
+import AjaxPromise from "stock/utils/ajax-promise"; //jshint ignore:line
 
 export default detail.extend({
-  setupController(controller, model){
-    if(controller) {
-      this._super(controller, model);
-      controller.set('isActiveSummary', true);
+  async model() {
+    const order = await this._super(...arguments);
+    const userId = order.get("createdBy.id");
+
+    return Ember.RSVP.hash({
+      order,
+      ordersCount: await new AjaxPromise(
```

Comment:
We should move all those loose and wild api calls into services, we really shouldn't be doing that this way, it's not maintainable and it clutters the code.

--> not really a comment on this code specifically, just saw an opportunity to mention it. 
If anyone has an opportunity to take the lead in this change wants to create those services, you have my blessing 🤗 (also do drop me a message if you wish to get involved, I have a few tips)

cc - @abulsayyad123 @namrataukirde @Darshan-Chauhan @abhinav098 @steveyken

Reply:
Yes.. that would be good thing to consider. Angular 2+ also, wraps all there ajax request request into singleton services.

Reply:
yes that seems like a very sensible idea. wrapping into a service provides a single place to maintain the code calls e.g. if the API changes or it's required elsewhere. Also simplifies testing :)